bytes.go

```diff
@@ -77,30 +73,33 @@ func ToMegabytes(s string) (uint64, error) {
 // GB = G = GiB = 1024 * M
 // TB = T = TiB = 1024 * G
 func ToBytes(s string) (uint64, error) {
-	parts := bytesPattern.FindStringSubmatch(strings.TrimSpace(s))
-	if len(parts) < 3 {
-		return 0, invalidByteQuantityError
-	}
+	s = strings.TrimSpace(s)
+	s = strings.ToUpper(s)
 
-	value, err := strconv.ParseFloat(parts[1], 64)
-	if err != nil || value <= 0 {
-		return 0, invalidByteQuantityError
-	}
+	i := strings.IndexFunc(s, func(v rune) bool {
+		return unicode.IsLetter(v)
+	})
 
-	var bytes uint64
-	unit := strings.ToUpper(parts[2])
-	switch unit[:1] {
-	case "T":
-		bytes = uint64(value * TERABYTE)
-	case "G":
-		bytes = uint64(value * GIGABYTE)
-	case "M":
-		bytes = uint64(value * MEGABYTE)
-	case "K":
-		bytes = uint64(value * KILOBYTE)
-	case "B":
-		bytes = uint64(value * BYTE)
+	if i != -1 {
```

Comment:
can you flip this test. I personally prefer 
```
if i == -1 {
  return error
}
continue processing
```

over 

```
if i != -1 {
  continue processing
}
return error
```

Reply:
this will complicate the code for understanding, I think, and add code dubbing

Reply:
Before:

```go
func ToBytes(s string) (uint64, error) {
	s = strings.TrimSpace(s)
	s = strings.ToUpper(s)

	i := strings.IndexFunc(s, unicode.IsLetter)

	if i != -1 {
		bytesString, multiple := s[:i], s[i:]
		bytes, err := strconv.ParseFloat(bytesString, 64)
		if err != nil || bytes <= 0 {
			return 0, invalidByteQuantityError
		}

		switch multiple {
		case "T", "TB", "TIB":
			return uint64(bytes * TERABYTE), nil
		case "G", "GB", "GIB":
			return uint64(bytes * GIGABYTE), nil
		case "M", "MB", "MIB":
			return uint64(bytes * MEGABYTE), nil
		case "K", "KB", "KIB":
			return uint64(bytes * KILOBYTE), nil
		case "B":
			return uint64(bytes), nil
		}
	}

	return 0, invalidByteQuantityError
}
```

After:
```go
func ToBytes(s string) (uint64, error) {
	s = strings.TrimSpace(s)
	s = strings.ToUpper(s)

	i := strings.IndexFunc(s, unicode.IsLetter)

	if i == -1 {
		return 0, invalidByteQuantityError
	}

	bytesString, multiple := s[:i], s[i:]
	bytes, err := strconv.ParseFloat(bytesString, 64)
	if err != nil || bytes <= 0 {
		return 0, invalidByteQuantityError
	}

	switch multiple {
	case "T", "TB", "TIB":
		return uint64(bytes * TERABYTE), nil
	case "G", "GB", "GIB":
		return uint64(bytes * GIGABYTE), nil
	case "M", "MB", "MIB":
		return uint64(bytes * MEGABYTE), nil
	case "K", "KB", "KIB":
		return uint64(bytes * KILOBYTE), nil
	case "B":
		return uint64(bytes), nil
	default:
		return 0, invalidByteQuantityError
	}
}
```

What you think?

Reply:
I think that's fine. I prefer reducing nesting and the duplication isn't any worse than it used to be before.

Reply:
kk

Reply:
fixed